### PR TITLE
test(WMG): 5050 Conditionally run WMG e2e tests based on API URL

### PR DESCRIPTION
**Pull request by tihuan**
## Reason for Change

- #5050 
- If the reason for this PR's code changes are not clear in the issue, state value/impact

## Changes

- add
- remove
- modify

## Testing steps

- Either list QA steps or reasoning you feel QA is unnecessary
- Describe how you made sure to know that your changes worked. Should allow someone else to go verify your code without in depth knowledge.
- "Unit tested only", "tested in rdev by a, b, c, verifying feature worked by... ", "manually ran pipeline locally with these results: ..."

### CODE REVIEW

File: frontend/playwright-globalSetup.ts [hunk 1]
```diff
@@ -2,23 +2,32 @@ import { chromium } from "@playwright/test";
...
+  const browserContext = await browser.newContext({
+    ...COMMON_PLAYWRIGHT_CONTEXT,
+    storageState: featureFlags,
+  });
+
+  const page = await browserContext.newPage();
+
+  await goToPage(undefined, page);
+
+  const apiUrl = await page.evaluate(() => {
```
Comment (tihuan) [inline frontend/playwright-globalSetup.ts:19]: during global setup phase of Playwright, we hit the homepage to detect which API URL the FE app connects to and save that as env variable for subsequent tests to access via helper function `conditionallyRunTests()`

File: frontend/tests/utils/wmgUtils.ts [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
 
+const { skip } = test;
+
+const ENVS_TO_RUN_TESTS = [
+  "api.cellxgene.dev.single-cell.czi.technology",
+  "api.cellxgene.staging.single-cell.czi.technology",
+  "api.cellxgene.cziscience.com",
+];
+
+export function conditionallyRunTests() {
```
Comment (tihuan) [inline frontend/tests/utils/wmgUtils.ts:16]: this is the function that will use the env variable and check if the API URL is one of the URLs in `ENVS_TO_RUN_TESTS` to determine whether to run the tests or not

File: frontend/tests/utils/helpers.ts [hunk 3]
```diff
@@ -57,11 +57,13 @@ const TEST_ENVS_DEV_STAGING = ["dev", "staging"];
...
 export const isDevStaging = TEST_ENVS_DEV_STAGING.includes(TEST_ENV);
 
+const GO_TO_PAGE_TIMEOUT_MS = 30 * 1000;
+
 export async function goToPage(
   url: string = TEST_URL,
   page: Page
 ): Promise<void> {
-  await page.goto(url);
+  await page.goto(url, { timeout: GO_TO_PAGE_TIMEOUT_MS });
```
Comment (tihuan) [inline frontend/tests/utils/helpers.ts:66]: default 30 seconds is not enough in GHA

**Review by atarashansky — APPROVED** [review]
Very nifty. Thanks for this!

**Comment by tihuan** [comment]
Thanks for the quick review, @atarashansky 🎉 !

**Comment by tihuan** [comment]
Ah the test is failing for real, because we've updated the WMG cubes, and since the test selects the first dataset, which is now `22 integrated samples`, it doesn't narrow down the options. 

I'll update the test to select disease `acute kidney failure` on Tuesday, so our test is more predictable and not dependent on datasets order, which changes more frequently 👌

**Comment by tihuan** [comment]
Fixed the test 👍

**Pull request merged**